plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/function/tablechanges/DeltaLakeTableChangesFunction.java

```diff
@@ -0,0 +1,190 @@
+            TransactionLogAccess transactionLogAccess,
+            TypeManager typeManager,
+            CachingExtendedStatisticsAccess statisticsAccess,
+            TrinoFileSystemFactory fileSystemFactory)
+    {
+        super(
+                SCHEMA_NAME,
+                NAME,
+                ImmutableList.of(
+                        ScalarArgumentSpecification.builder().name(TABLE_NAME_ARGUMENT).type(VARCHAR).build(),
+                        ScalarArgumentSpecification.builder().name(START_VERSION_ARGUMENT).type(BIGINT).defaultValue(-1L).build(),
+                        ScalarArgumentSpecification.builder().name(END_VERSION_ARGUMENT).type(BIGINT).defaultValue(Long.MAX_VALUE).build()),
+                GENERIC_TABLE);
+        this.hiveMetastoreFactory = requireNonNull(hiveMetastoreFactory, "hiveMetastoreFactory is null");
+        this.transactionLogAccess = requireNonNull(transactionLogAccess, "transactionLogAccess is null");
+        this.typeManager = requireNonNull(typeManager, "typeManager is null");
+        this.statisticsAccess = requireNonNull(statisticsAccess, "statisticsAccess is null");
+        this.fileSystemFactory = requireNonNull(fileSystemFactory, "fileSystemFactory is null");
+    }
+
+    @Override
+    public TableFunctionAnalysis analyze(ConnectorSession session, CatalogHandle catalogHandle, ConnectorTransactionHandle transaction, Map<String, Argument> arguments)
+    {
+        ScalarArgument tableNameArgument = (ScalarArgument) arguments.get(TABLE_NAME_ARGUMENT);
+        requireNonNull(tableNameArgument.getValue(), "tableName value for function table_changes() is null");
+        String schemaTableName = ((Slice) tableNameArgument.getValue()).toStringUtf8();
+        if (!schemaTableName.contains(".") || schemaTableName.indexOf(".") != schemaTableName.lastIndexOf(".")) {
+            throw new IllegalArgumentException("table_name value for function table_changes() must be in following format: schema_name.table_name");
+        }
+
+        ScalarArgument startVersionArgument = (ScalarArgument) arguments.get(START_VERSION_ARGUMENT);
+        Long startVersion = (Long) startVersionArgument.getValue();
+        requireNonNull(startVersion, "startVersion value for function table_changes() is null");
+        startVersion++; // to guarantee exclusiveness for the lower end of the range
+        checkArgument(startVersion >= 0, "startVersion value for function table_changes() must not be negative");
+
+        ScalarArgument endVersionArgument = (ScalarArgument) arguments.get(END_VERSION_ARGUMENT);
+        Long endVersion = (Long) endVersionArgument.getValue();
+        requireNonNull(endVersion, "endVersion value for function table_changes() is null");
+        checkArgument(endVersion >= 0, "endVersion value for function table_changes() must not be negative");
+        checkArgument(endVersion >= startVersion, "endVersion value for function table_changes() must be greater than startVersion");
```

Comment:
I think it'd be better to throw a TrinoException with the `INVALID_TABLE_FUNCTION_INVOCATION` code since it's a user error, not a code bug